providers/os/connection/container/image_connection_test.go

```diff
@@ -68,6 +70,32 @@ type dockerConnTest struct {
 	testfile string
 }
 
+func TestNewImageConnection_DelayDiscovery(t *testing.T) {
+	ref, err := name.ParseReference(alpineImage, name.WeakValidation)
+	require.NoError(t, err)
+
+	img, err := image.LoadImageFromRegistry(ref)
+	require.NoError(t, err)
+
+	inv := &inventory.Config{Options: map[string]string{}}
+	_, err = container.NewImageConnection(1, inv, &inventory.Asset{}, img)
+	require.NoError(t, err)
+	assert.True(t, inv.DelayDiscovery)
```

Comment:
Could we add a check for the platform here and in the other test?

Here, we shouldn't have any platform information. In the other test we should know it is an alpine image.